extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkStore.java

```diff
@@ -84,7 +84,7 @@ void writeToEntities(EntityManager session, PolarisBaseEntity entity) {
     diagnosticServices.check(session != null, "session_is_null");
     checkInitialized();
 
-    ModelEntity model = lookupEntity(session, entity.getCatalogId(), entity.getId());
+    ModelEntity model = lookupEntity(session, entity.getCatalogId(), entity.getId(), -1);
```

Comment:
Should we use NULL_TYPE here? Some new construct like UNKNOWN_TYPE / ANY_TYPE? Make the type argument Optional?

Reply:
Good question, my idea with -1 was precisely to indicate the absence of type code, since there is no such type code in `PolarisEntityType`.

I would be OK with all your suggestions. Let's wait a bit for other people weigh in.

Reply:
I will admit I assumed that, but I had to go check to make sure -1 is not a real code :)

I thought one of the other constructs might be safer. We could even keep -1 and just call it ANY_TYPE. Anyway, let's see what others think!

Reply:
Why not use `entity.getTypeCode()`?

Reply:
+1 to using that if it's applicable. My comment is more about what we should do in the case that we _don't_ want to filter by type code, or we don't have a type code available. -1 seems like an okay way to represent this but I wanted to explore alternatives.

Reply:
> Why not use entity.getTypeCode()?

Because I'm not sure it makes sense on write paths. Here we are looking up an existing entity to decide whether the write is an update or an insert. So the type of the existing entity actually doesn't matter much, since it will end up overwritten by the new one.